src/main/java/com/softserveinc/dokazovi/service/impl/UserServiceImpl.java

```diff
@@ -97,7 +97,11 @@ public Page<UserEntity> findAll(Pageable pageable) {
      */
     @Override
     public UserDTO findExpertById(Integer userId) {
-        return userMapper.toUserDTO(userRepository.findById(userId).orElse(null));
+        AuthorEntity author = authorRepository.findById(userId).orElse(null);
+        if (author == null) {
+            throw new EntityNotFoundException("Author not found");
+        }
+        return userMapper.toUserDTO(userRepository.findById(author.getProfile().getId()).orElse(null));
```

Comment:
It's not a good practice to return null value. Try to avoid such cases if possible.

Reply:
It's definitely better than returning null.